datafusion/sqllogictest/test_files/string/string_literal.slt

```diff
@@ -15,6 +15,15 @@
 # specific language governing permissions and limitations
 # under the License.
 
+# a backslash in a string literal is not a special character
+query T
+VALUES ('\'), ('\\'), ('\\\'), ('\\\\')
+----
+\
+\\
+\\\
+\\\\
```

Comment:
This query (and all other queries involving a `\` backslash) would behave differently in the CLI
https://github.com/apache/datafusion/issues/13286